Document/0x06e-Testing-Cryptography.md

```diff
@@ -2,12 +2,46 @@
 
-### iOS Cryptography Libraries
+### Verifying the Configuration of Cryptographic Standard Algorithms
 
+#### Overview
 Apple provides libraries that include implementations of most common cryptographic algorithms. [Apple's Cryptographic Services Guide](https://developer.apple.com/library/content/documentation/Security/Conceptual/cryptoservices/GeneralPurposeCrypto/GeneralPurposeCrypto.html "Apple Cryptographic Services Guide") is a great reference. It contains generalized documentation of how to use standard libraries to initialize and use cryptographic primitives, information that is useful for source code analysis.
 
-iOS code usually refers to constants defined in `CommonCryptor.h` (for example, `kCCAlgorithmDES`). You can search the source code for these constants to detect their use. Because iOS constants are numeric, you should determine whether the constants sent to the `CCCrypt` function represent an insecure or deprecated algorithm.
-
+##### CommonCrypto, SecKeyEncrypt and Wrapper libraries
+The most commonly used Class for cryptographic operations is the CommonCrypto, which is packed with the iOS runtime. The functionality offered by the CommonCrypto object can best be disected by having a look at the [source code of the header file ](https://opensource.apple.com/source/CommonCrypto/CommonCrypto-36064/CommonCrypto/CommonCrypto.h "CommonCrypto.h"):
+- The `Commoncryptor.h` gives the parameters for the symmetric cryptographic operations,
+- The `CommonDigest.h` gives the parameters for the hashing Algorithms
+- The `CommonHMAC.h` gives the parameters for the supported HMAC operations.
+- The `CommonKeyDerivation.h` gives the parameters for supported KDF functions
+- The `CommonSymmetricKeywrap.h` gives the function used for wrappnig a symmetric key with a Key Encryption Key.
+
+CommonCryptor lacks a few type of operations unfortunately in its public APIs, for instance: GCM mode is only available in its private APIs: see [its sourcecode](https://opensource.apple.com/source/CommonCrypto/CommonCrypto-60074/include/CommonCryptorSPI.h "GCM in CC"). For this, an additional binding header is necessary or other wrapper libraries can be used.
+
+Next, for asymmetric operations, Apple provides [SecKey](https://opensource.apple.com/source/Security/Security-57740.51.3/keychain/SecKey.h.auto.html "SecKey"). Apple provides a nice guide in its [Developer Documentation](https://developer.apple.com/documentation/security/certificate_key_and_trust_services/keys/using_keys_for_encryption?language=objc "Using keys for encryption") on how to use this.
```

Comment:
Parameter "language" in the link sets the default language for code-snippets to Objective-C. If this is intended : no problem. if not : remove it for defaulting to Swift

Reply:
it is often intended, but i will check which ones should be in Swift.

Reply:
all done 👍